src/class.c

```diff
@@ -1634,22 +1661,27 @@ bool InterfaceDeclaration::isBaseOf(ClassDeclaration *cd, int *poffset)
     {
         BaseClass *b = cd->interfaces[j];
 
-        //printf("\tbase %s\n", b->base->toChars());
-        if (this == b->base)
+        //printf("\tX base %s\n", b->sym->toChars());
+        if (this == b->sym)
         {
             //printf("\tfound at offset %d\n", b->offset);
             if (poffset)
             {
                 *poffset = b->offset;
-                if (j && cd->isInterfaceDeclaration())
+                // The conversion is interface(cd) to interface(this).
+                if (cd->isInterfaceDeclaration())
```

Comment:
> excplitely just casting to Object for comparison is faster and explicite.

A cast from interface to class/interface is done with the combination of:
1. A conversion from D interface to Object, it is a trivial pointer operation.
2. Object to class/interface cast may be cross cast, so it requires class hierarchy search.

Are you saying about the (1)? If so, it's already done in `_d_interface_cast` function.